plugins/inputs/intel_pmt/filtering.go

```diff
@@ -0,0 +1,269 @@
+		}
+	}
+	a.SampleGroup = newSampleGroup
+}
+
+// filterAggInterfaceByDatatype filter aggregator interface XML by provided datatypes.
+//
+// Aggregator interface XML contains many aggregator samples inside, each with datatype assigned.
+// This function filters aggregator samples based on their datatype.
+//
+// Parameters:
+//
+//	datatypes: string slice of datatypes to include in filtered XML.
+//	dtMetricsFound: a map of found datatypes for all GUIDs.
+func (a *aggregatorInterface) filterAggInterfaceByDatatype(datatypes []string, dtMetricsFound map[string]bool) {
+	newAggSample := aggregatorSamples{}
+	for _, s := range a.AggregatorSamples.AggregatorSample {
+		if slices.Contains(datatypes, s.DatatypeIDRef) {
+			dtMetricsFound[s.DatatypeIDRef] = true
+			newAggSample.AggregatorSample = append(newAggSample.AggregatorSample, s)
+		}
+	}
+	a.AggregatorSamples = newAggSample
+}
+
+// filterAggInterfaceBySampleName filters aggregator interface XML by sample names.
+//
+// This function filters aggregator samples based on the provided sampleNames.
+// When the name for the sample is unique the match is exact.
+// When the name is per resource (i.e. Cx_) the match is regex based.
+//
+// Parameters:
+//
+//	sampleNames: string slice of sample names to include in filtered XML.
+//	sMetricsFound: a map of found metric names for all GUIDs.
+func (a *aggregatorInterface) filterAggInterfaceBySampleName(sampleNames []string, sMetricsFound map[string]bool) {
+	newAggSample := aggregatorSamples{}
+aggInterfaceLoop:
+	for _, s := range a.AggregatorSamples.AggregatorSample {
+		for _, userMetricInput := range sampleNames {
+			if strings.Contains(s.SampleName, userMetricInput) {
+				newAggSample.AggregatorSample = append(newAggSample.AggregatorSample, s)
+				sMetricsFound[s.SampleName] = true
+				continue aggInterfaceLoop
+			}
+			for patternInput, re := range metricPatterns {
+				if strings.EqualFold(userMetricInput, patternInput) {
+					if re.MatchString(s.SampleName) {
+						newAggSample.AggregatorSample = append(newAggSample.AggregatorSample, s)
+						sMetricsFound[s.SampleName] = true
+						continue aggInterfaceLoop
+					}
+				}
+			}
+		}
+	}
+	a.AggregatorSamples = newAggSample
+}
+
+// Patterns for some metrics, as they have CHAx or Cx in their name
+// with "x" being the core/cha number.
+// Inline with metrics described in readme.
+var metricPatterns = map[string]*regexp.Regexp{
+	// datatype: ttemperature
+	"Cx_TEMP": regexp.MustCompile(`C\d+_TEMP$`),
+
+	// datatype: tcore_state
+	"Cx_EN": regexp.MustCompile(`C\d+_EN$`),
+
+	// datatype: thist_counter
+	"Cx_FREQ_HIST_R0":  regexp.MustCompile(`C\d+_FREQ_HIST_R0$`),
+	"Cx_FREQ_HIST_R1":  regexp.MustCompile(`C\d+_FREQ_HIST_R1$`),
+	"Cx_FREQ_HIST_R2":  regexp.MustCompile(`C\d+_FREQ_HIST_R2$`),
+	"Cx_FREQ_HIST_R3":  regexp.MustCompile(`C\d+_FREQ_HIST_R3$`),
+	"Cx_FREQ_HIST_R4":  regexp.MustCompile(`C\d+_FREQ_HIST_R4$`),
+	"Cx_FREQ_HIST_R5":  regexp.MustCompile(`C\d+_FREQ_HIST_R5$`),
```

Comment:
We could use this regex on all samples, capture the `var` group, compare this group to what the user has provided as `sample_filter` and append if it is in the provided slice.
I don't know if this is what you meant but this would remove the need for this map.

Reply:
Removed the entire map, using only one regex now to capture the `var` group.